src/NHibernate/Linq/DefaultQueryProvider.cs

```diff
@@ -78,15 +78,14 @@ public virtual object ExecuteFutureAsync(Expression expression, CancellationToke
 
 		public async Task<TResult> ExecuteAsync<TResult>(Expression expression, CancellationToken cancellationToken)
 		{
-			return (TResult)await ExecuteAsync(expression, cancellationToken);
+			return (TResult)await ExecuteAsync(expression, cancellationToken).ConfigureAwait(false);
 		}
 
-		public virtual async Task<object> ExecuteAsync(Expression expression, CancellationToken cancellationToken)
+		public virtual Task<object> ExecuteAsync(Expression expression, CancellationToken cancellationToken)
 		{
-			IQuery query;
-			NhLinqExpression nhLinqExpression = PrepareQuery(expression, out query);
+			NhLinqExpression nhLinqExpression = PrepareQuery(expression, out var query);
 
-			return await ExecuteQueryAsync(nhLinqExpression, query, nhLinqExpression, cancellationToken);
```

Comment:
By removing the async/await keywords the exception propagation is [now different](https://stackoverflow.com/a/23386886). If there is an exception inside the `PrepareQuery` method the exception will be immediately propagated to the called instead of returning a task that contains the exception. Currently, we are throwing a raw exception only for [preconditions](https://github.com/maca88/AsyncGenerator/issues/1). IMO I would wrap the method inside a try/catch block like the generator would do:
```
public virtual Task<object> ExecuteAsync(Expression expression, CancellationToken cancellationToken)
{
    try
    {
        IQuery query;
        NhLinqExpression nhLinqExpression = PrepareQuery(expression, out query);

        return ExecuteQueryAsync(nhLinqExpression, query, nhLinqExpression, cancellationToken);
    }
    catch (Exception ex)
    {
        return Task.FromException<object>(ex);
    }
}
```